### PR TITLE
Prefer attributes, deprecate annotations

### PR DESCRIPTION
**Pull request by fulldecent**

### CODE REVIEW

**Comment by fulldecent** [comment]
This adds a deprecation notice so we can soon update to only supported PHP versions.

---

For reference, [here are the supported versions of PHP](https://www.php.net/supported-versions.php).

Please note that future versions of PHP will have longer support cycles (just for the security support, not the active support), however 7.4 and even 8.0 are specifically excluded.

This is a clear signal to the community not to use 7.4 and 8.0.

File: README.md [hunk 1]
```diff
@@ -4,11 +4,12 @@
...
-[doctrine annotations](https://www.doctrine-project.org/projects/annotations.html) (optional as of version 4.8; if required the `doctrine/annotations` library must be installed in addition to swagger.php).
-or [PHP attributes](https://www.php.net/manual/en/language.attributes.overview.php).
+Generate interactive [OpenAPI](https://www.openapis.org) documentation for your RESTful API using [PHP attributes](https://www.php.net/manual/en/language.attributes.overview.php) (preferred) or 
+[doctrine annotations](https://www.doctrine-project.org/projects/annotations.html) (requires additional `doctrine/annotations` library).
 
-For a full list of supported annotations, please have look at the [`OpenApi\Annotations` namespace](src/Annotations) or the [documentation website](https://zircote.github.io/swagger-php/guide/annotations.html).
+See the [documentation website](https://zircote.github.io/swagger-php/guide/attributes.html) for supported attributes and annotations.
+
+Annotations, and support for PHP versions lower than 8.1 are deprecated and will be removed in a future release of swagger-php.
```
Comment (DerManoMann) [inline README.md:12]: Not quite sure about this one, in particular referencing a specific PHP version.
There are already two mentions of 8.1 as the min. required PHP version for attributes, so this doesn't really add anything.

`may be removed` is strong enough here, seeing that annotations are still widely used, IMO. If this changes faster than expected we can still change it.


```suggestion
Annotations are deprecated and may be removed in a future release of swagger-php.
```

**Review by DerManoMann — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (fulldecent) [reply README.md]: Thank you, baked in

File: README.md [hunk 2]
```diff
@@ -19,6 +20,8 @@ For a full list of supported annotations, please have look at the [`OpenApi\Anno
 - Exceptional error reporting (with hints, context)
 - As of PHP 8.1 all annotations are also available as PHP attributes
 
+
```
Comment (DerManoMann) [inline README.md:23]: oh, another nit: why these extra lines?

**Review by DerManoMann — CHANGES_REQUESTED** [review]
Sorry to be a pain...

File: README.md (hunk 2 above)
Comment (fulldecent) [reply README.md:23]: ugggh prettier, that's why :-(

**Comment by fulldecent** [comment]
Squash rebase, fix whitespace 

@DerManoMann

File: README.md (hunk 2 above)
Comment (fulldecent) [reply README.md]: did rebase squash should be fixed now

**Review by DerManoMann — APPROVED** [review]
I do not particularly like the `(preferred)` - the library doesn't really care ;) I suppose it is the more modern approach :shrimp:

**Comment by DerManoMann** [comment]
Thanks @fulldecent

**Pull request merged**